config-linux.h: fix glibc build failure

Recent glibc v2.30 dropped XSI STREAMS declarations,
which causing below build failure.

poky/build/tmp/work/corei7-64-poky-linux/ace/6.5.3-r0/ACE_wrappers/ace/os_include/os_stropts.h:56:17: fatal error: stropts.h: No such file or directory
|    56 | #  include /**/ <stropts.h>
|       |                 ^~~~~~~~~~~
| compilation terminated.

Added GLIBC checks for affected versions.

For more information about glibc v2.30 change, please check:
https://sourceware.org/git/?p=glibc.git;a=commit;h=a0a0dc83173ce11ff45105fd32e5d14356cdfb9c

Upstream-Status: Submitted [https://github.com/DOCGroup/ACE_TAO/pull/939]

Signed-off-by: Naveen Saini <naveen.kumar.saini@intel.com>

--- ACE_wrappers_original/ace/config-linux.h	2018-11-19 17:45:53.163139000 +0800
+++ ACE_wrappers/ace/config-linux.h	2019-08-13 17:04:17.620866464 +0800
@@ -295,7 +295,7 @@
 
 // Starting with FC9 rawhide this file is not available anymore but
 // this define is set
-#if defined _XOPEN_STREAMS && _XOPEN_STREAMS == -1
+#if (defined(__GLIBC__) && (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 30)) || (defined _XOPEN_STREAMS && _XOPEN_STREAMS == -1)
 # define ACE_LACKS_STROPTS_H
 # define ACE_LACKS_STRRECVFD
 #endif
